src/app/domain/robot.type.ts

```diff
@@ -16,20 +19,24 @@ const DEFAULTS: Features = {
     showCodeOnStart: false
 };
 
+interface ProtocolInformation {
+    protocol: typeof BaseProtocol;
+    microcontroller?: string;
+}
+
 export class RobotType {
     public features: Features;
 
     constructor(
         public id: string,
-        public protocol: typeof BaseProtocol,
+        public protocol: Partial<ProtocolInformation>,
```

Comment:
Partial doesn't need to be used and removing can result in less if checks since the microcontroller param is already optional in the type